contracts/test/integration/IntegrationTestVanillaPSM.t.sol

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {MockPCVDepositV2} from "../../mock/MockPCVDepositV2.sol";
+import {IPCVDeposit} from "../../pcv/IPCVDeposit.sol";
+import {OraclePassThrough} from "../../oracle/OraclePassThrough.sol";
+import {ScalingPriceOracle} from "../../oracle/ScalingPriceOracle.sol";
+import {MockScalingPriceOracle} from "../../mock/MockScalingPriceOracle.sol";
+import {ICore} from "../../core/ICore.sol";
+import {Core} from "../../core/Core.sol";
+import {IVolt, Volt} from "../../volt/Volt.sol";
+import {BasePSM} from "../../peg/BasePSM.sol";
+import {IBasePSM} from "../../peg/IBasePSM.sol";
+import {VanillaPriceBoundPSM} from "../../peg/VanillaPriceBoundPSM.sol";
+import {VanillaPSM} from "../../peg/VanillaPSM.sol";
+import {getCore, getMainnetAddresses, VoltTestAddresses} from "../unit/utils/Fixtures.sol";
+import {IPCVDeposit} from "../../pcv/IPCVDeposit.sol";
+import {Vm} from "./../unit/utils/Vm.sol";
+import {DSTest} from "./../unit/utils/DSTest.sol";
+import {MainnetAddresses} from "./fixtures/MainnetAddresses.sol";
+import {Constants} from "../../Constants.sol";
+
+contract IntegrationTestVanillaPSMTest is DSTest {
+    using SafeCast for *;
+
+    VanillaPriceBoundPSM private priceBoundPsm;
+    VanillaPSM private vanillaPsm;
+
+    ICore private core = ICore(MainnetAddresses.CORE);
+    IVolt private volt = IVolt(MainnetAddresses.VOLT);
+    IERC20 private usdc = IERC20(MainnetAddresses.USDC);
+    IERC20 private underlyingToken = usdc;
+
+    address public makerUSDCPSM = MainnetAddresses.MAKER_USDC_PSM;
+
+    uint256 public constant mintAmount = 10_000_000e6;
+    uint256 public constant voltMintAmount = 10_000_000e18;
+
+    /// @notice live FEI PCV Deposit
+    IPCVDeposit public immutable pcvDeposit =
+        IPCVDeposit(MainnetAddresses.VOLT_USDC_PSM);
+
+    /// @notice Oracle Pass Through contract
+    OraclePassThrough public oracle =
+        OraclePassThrough(MainnetAddresses.ORACLE_PASS_THROUGH);
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    uint128 voltFloorPrice = 900_000;
+    uint128 voltCeilingPrice = 105_0000;
```

Comment:
we should make this ceiling price higher so it takes longer for the tests to start failing